packages/textarea/src/TextAreaRoot.tsx

```diff
@@ -28,12 +30,11 @@ export const TextAreaRoot = (StyledTextAreaRoot: any) =>
         isRequired: props.isRequired,
         nativeID: props.nativeID,
       });
-
       return (
         <StyledTextAreaRoot
           states={{
-            hover: isHovered,
-            focus: isFocused,
+            hover: isHovered ? isHovered : isHoveredVal,
+            focus: isFocusedVal ? isFocusedVal : isFocused,
```

Comment:
replace `isHovered ? isHovered : isHoveredVal` with `isHovered || isHoveredVal` for code consistency.